src/VolumeRenderSettings.ts

```diff
@@ -0,0 +1,141 @@
+import { Euler, Vector2, Vector3 } from "three";
+import Volume from "./Volume";
+import { Bounds } from "./types";
+
+/**
+ * Marks groups of related settings that may have changed.
+ */
+export enum SettingsFlags {
+  /** parameters: translation, rotation, scale, currentScale, flipAxes */
+  TRANSFORM = 1,
+  /** parameters: gammaMin, gammaLevel, gammaMax, brightness*/
+  CAMERA = 2,
+  /** parameters: showBoundingBox, boundingBoxColor */
+  BOUNDING_BOX = 4,
+  /** parameters: bounds, zSlice */
+  ROI = 8,
+  /** parameters: maskChannelIndex, maskAlpha */
+  MASK = 16,
+  /** parameters: density, specular, emissive, glossiness */
+  MATERIAL = 32,
+  /** parameters: resolution, useInterpolation, pixelSamplingRate, primaryRayStepSize, secondaryRayStepSize*/
+  SAMPLING = 64,
+  /** parameters: isOrtho, orthoScale, viewAxis, visible, maxProjectMode */
+  VIEW = 128,
+  ALL = Number.MAX_SAFE_INTEGER,
+}
+
+export enum Axis {
+  X = "x",
+  Y = "y",
+  Z = "z",
+  NONE = "",
+}
+
+/**
+ * Holds shared settings for configuring `VolumeRenderImpl` instances.
+ */
+export type VolumeRenderSettings = {
+  // TRANSFORM
+  translation: Vector3;
+  rotation: Euler;
+  scale: Vector3;
+  currentScale: Vector3;
+  flipAxes: Vector3;
+
+  // VIEW
+  isOrtho: boolean;
+  orthoScale: number;
+  viewAxis: Axis;
+  visible: boolean;
+  maxProjectMode: boolean;
+
+  // CAMERA
+  gammaMin: number;
+  gammaLevel: number;
+  gammaMax: number;
+  brightness: number;
+
+  // MASK
+  maskChannelIndex: number;
+  maskAlpha: number;
+
+  // MATERIAL
+  density: number;
+  specular: [number, number, number][];
+  emissive: [number, number, number][];
+  glossiness: number[];
+
+  // ROI
+  bounds: Bounds;
+  zSlice: number;
+
+  // BOUNDING_BOX
+  showBoundingBox: boolean;
+  boundingBoxColor: [number, number, number];
+
+  // SAMPLING
+  resolution: Vector2;
+  useInterpolation: boolean;
+  pixelSamplingRate: number;
+  primaryRayStepSize: number;
+  secondaryRayStepSize: number;
+};
+
+/**
+ * Returns a VolumeRenderSettings object with default fields. Default objects
+ * created with this method will not have shared references.
+ *
+ * Note that default objects have volume-dependent properties that should be updated
+ * with `VolumeRenderSettingUtils.updateWithVolume()`.
```

Comment:
why not just pass a Volume in to this function?  do you ever call it by itself?

Reply:
This was a good suggestion, thank you! Added :)